apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedissonMethodInterceptor.java

```diff
@@ -45,14 +44,11 @@
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                              MethodInterceptResult result) throws Throwable {
         RedisConnection connection = (RedisConnection) objInst;
+        Channel channel = connection.getChannel();
         String peer = "";
-        if (connection.getRedisClient() != null) {
-            Field field = RedisClient.class.getDeclaredField("uri");
-            field.setAccessible(true);
-            URI uri = (URI) field.get(connection.getRedisClient());
-            if (uri != null) {
-                peer = uri.getHost() + ":" + uri.getPort();
-            }
+        if (channel.remoteAddress() instanceof InetSocketAddress) {
```

Comment:
Is there any way to cache this peer info? Build string and instanceof are not very efficiency, especially in jre 6 and 7.

Reply:
Such as using dynamic field.

Reply:
'channel.remoteAddress() instanceof InetSocketAddress',in order to reduce cast exception，netty channel remoteAddress most InetSocketAddress。
i think cache peer is very difficult in redisson non-standalone mod, because I don't know which remote resource netty will use.

Reply:
i think using dynamic field will not know where the specific action occurred on that node.

Reply:
Action could set in every access. But peer don't need. Then you could avoid instanceof in every time

Reply:
I am not saying instanceof is not allowed. I just suggest you don't use it in every span building phase